enigma/src/main/java/org/quiltmc/enigma/impl/plugin/RecordIndexingVisitor.java

```diff
@@ -72,15 +88,22 @@ public MethodVisitor visitMethod(final int access, final String name, final Stri
 	public void visitEnd() {
 		super.visitEnd();
 		try {
-			if (this.clazz != null) {
-				this.collectResults();
-			}
+			this.collectResults();
+
+			this.clazz = null;
+			this.recordComponents.clear();
+			this.fields.clear();
+			this.methods.clear();
```

Comment:
please double-check that moving the clearing here and clearing fields+methods won't break anything